app/src/test/js/keysCtrlTest.js

```diff
@@ -0,0 +1,158 @@
+
+            var syncKeysCtrl, searchKeysCtrl, http, globals, scope, rootScope, esbMessageService, filter;
+
+            beforeEach(inject(function($rootScope, $controller, $httpBackend,
+                    Globals, EsbMessageService, $filter) {
+                rootScope = $rootScope;
+                syncKeysScope = $rootScope.$new();
+                searchKeysScope = $rootScope.$new();
+                http = $httpBackend;
+                globals = Globals;
+                esbMessageService = EsbMessageService;
+                filter = $filter;
+
+                (syncKeysCtrl) = $controller('SyncKeysCtrl', {
+                    '$scope' : syncKeysScope
+                });
+                http.expectGET("api/key/tree/Entities").respond(
+                        syncKeysSuccessResponse);
+                http.flush();
+
+                (searchKeysCtrl) = $controller('SearchKeysCtrl', {
+                    '$scope' : searchKeysScope
+                });
+                http.expectGET("api/key/tree/SearchKeys").respond(
+                        searchKeysSuccessResponse);
+                http.flush();
+
+            }));
+
+            function loadTest(scope, response) {
+                expect(scope.entities || scope.searchKeys).toEqual(
+                        response.tree);
+                expect(scope.parent).toEqual(response.tree);
+            }
+
+            it('loads Sync and Search Keys Tab', function() {
+                loadTest(syncKeysScope, syncKeysSuccessResponse);
+                loadTest(searchKeysScope, searchKeysSuccessResponse);
+            });
+
+            function crumbTest(scope, response) {
+                expect(scope.crumbs.length).toEqual(1);
+                expect(scope.crumbs[0]).toEqual(response.tree);
+                expect(scope.parent).toEqual(response.tree);
+                scope.manageChildren(scope.parent.children[0]);
+
+                expect(scope.crumbs.length).toEqual(2);
+                expect(scope.crumbs[0]).toEqual(response.tree);
+                expect(scope.crumbs[1]).toEqual(response.tree.children[0]);
+                expect(scope.parent).toEqual(response.tree.children[0]);
+
+                scope.manageChildren(scope.parent.children[0]);
+                expect(scope.crumbs.length).toEqual(3);
+                expect(scope.crumbs[0]).toEqual(response.tree);
+                expect(scope.crumbs[1]).toEqual(response.tree.children[0]);
+                expect(scope.crumbs[2]).toEqual(
+                        response.tree.children[0].children[0]);
+                expect(scope.parent).toEqual(
+                        response.tree.children[0].children[0]);
+
+                scope.gotoCrumb(scope.crumbs[0]);
+                expect(scope.crumbs.length).toEqual(1);
+                expect(scope.crumbs[0]).toEqual(response.tree);
+                expect(scope.parent).toEqual(response.tree);
+            }
+
+            it('crumb and manage chidren tests', function() {
```

Comment:
"crumb and manage children tests" is pretty vague. Tests serve as documentation for the responsibilities of objects (controllers, in this case). This doesn't tell me much about these controllers. Specifically, I think the `manageChildren` and `gotoCrumb` methods probably deserve their own tests (and perhaps multiple tests for each to describe various behaviors given various scenarios).